From df107bae6b0283b591dd542e690d70be09c3475f Mon Sep 17 00:00:00 2001
From: Michel JAOUEN <michel.jaouen@st.com>
Date: Thu, 20 Jun 2024 09:12:47 +0200
Subject: [PATCH 1205/1695] product/tfm-stm32mp2: fix module clock

Since a null function is handled as a fatal error on a request,
add a get_rate_from_index function returning not supported.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 5bb585eb712b648257cd02e85f56f76ce308aa1f product/tfm-stm32mp2: fix module clock
9e82deb3494a7a162868353bd4a7ab5ae63ff0a2 product/tfm-stm32mp2: fix module clock]
Change-Id: Ifaf67a1e4fac0eef4deb08f610857b48438e034f
Signed-off-by: Michel JAOUEN <michel.jaouen@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/SCP-firmware/+/391976
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Etienne CARRIERE <etienne.carriere@foss.st.com>
Domain-Review: Etienne CARRIERE <etienne.carriere@foss.st.com>
---
 .../tfm-stm32mp2/module/tfm_clock/src/mod_tfm_clock.c     | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/core/lib/scmi-server/SCP-firmware/product/tfm-stm32mp2/module/tfm_clock/src/mod_tfm_clock.c b/core/lib/scmi-server/SCP-firmware/product/tfm-stm32mp2/module/tfm_clock/src/mod_tfm_clock.c
index e84ff30c4..d42eba2e0 100644
--- a/core/lib/scmi-server/SCP-firmware/product/tfm-stm32mp2/module/tfm_clock/src/mod_tfm_clock.c
+++ b/core/lib/scmi-server/SCP-firmware/product/tfm-stm32mp2/module/tfm_clock/src/mod_tfm_clock.c
@@ -298,12 +298,18 @@ static int stub_pending_power_transition(fwk_id_t dev_id,
     return FWK_E_SUPPORT;
 }
 
+static int get_rate_from_index(fwk_id_t dev_id,
+                               unsigned int rate_index, uint64_t *rate)
+{
+    return FWK_E_SUPPORT;
+}
+
 static const struct mod_clock_drv_api api_tfm_clock = {
     .get_rate = get_rate,
     .set_state = set_state,
     .get_state = get_state,
     .get_range = get_range,
-    .get_rate_from_index = NULL,
+    .get_rate_from_index = get_rate_from_index,
     .set_rate = set_rate,
     .get_duty_cycle = get_duty_cycle,
     .round_rate = get_rounded_rate,
-- 
2.47.2

